src/mpid/ch4/src/ch4_startall.h

```diff
@@ -21,6 +23,31 @@ MPL_STATIC_INLINE_PREFIX int MPID_Startall(int count, MPIR_Request * requests[])
         if (MPIDIG_REQUEST(preq, rank) == MPI_PROC_NULL)
             continue;
 
+        /* Persistent collectives support */
+        if (preq->kind == MPIR_REQUEST_KIND__PREQUEST_BCAST) {
```

Comment:
Do you envision more fine-grained control over different persistent collective request types? Would a single `MPIR_REQUEST_KIND__PREQUEST_COLL` type be sufficient?

Reply:
I think a single request kind may be sufficient.  I can't think of use case for specific requests. @zhenggb72 what do you think?

Reply:
I think so. We can just use MPIR_REQUEST_KIND__PREQUEST_COLL

Reply:
Is the PR updated with this discussion? Does it make this commit (https://github.com/pmodels/mpich/pull/4859/commits/1c8038912b1d2d6ad15c3e52de86d4e62bad59d1) unnecessary now?

Reply:
I think an increase in `MPIDI_REQUEST_TYPE_MASK` is still needed because the 16 values from previous mask are already in use. But it can probably change from  `F` -> `1F`  and not `3F`

Reply:
I see. It only affects `ch3`, right? If change is necessary, the I guess 0x3f and 0x1f don't make much difference.

Reply:
Yeah, its only in `ch3`. I will keep this commit then.